qiime/core/type/signature.py

```diff
@@ -94,10 +110,30 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    @property
+    def descriptions(self):
+        params = {**self.inputs, **self.parameters, **self.outputs}
```

Comment:
I think you lose order here when converting to a set, so the `OrderedDict` will have arbitrary order. I'm not sure how important order is here though. @ebolyen, do we need this method, or is it enough that each `ParameterSpec` has a `description`? I think the `defaults` property (above) was added for backwards-compatibility, so we may not need a corresponding property for descriptions.